### ISSUE TITLE
[Linux]: network actvitiy per process 

### ISSUE THREAD

**Issue by thibaultmol** [giampaolo/psutil#1900]
## Summary

* OS: Linux
* Type: core

## Description

{{{
  Right now it's not possible with psutil to see how much network activity each process uses. 
  Yet this is incredibly useful, for example: when you want to see which process is taking up all your pc's bandwidth
}}}

**Comment by giampaolo** [comment]
This is definitively a nice-to-have feature, but unfortunately it looks like per-process net IO counters are not exposed by the Linux kernel. There's `/proc/{PID}/net/netstat` and `/proc/{PID}/net/dev`, but they seem to provide system-wide (or [similar](https://unix.stackexchange.com/questions/224201/what-is-proc-pid-net-dev)) metrics, not per-process.

https://github.com/raboof/nethogs/ seem to be able to do this, but since I see it depends on `libpcaps` (a traffic sniffing library), I suppose that means nethogs starts collecting "cumulative" metrics only after it's started (EDIT: confirmed). That suggests (again) that the kernel doesn't keep track of these per-process counters anywhere.

**Comment by thibaultmol** [comment]
I wonder how ksysguard does it, (which is what i normally use)

**Comment by giampaolo** [comment]
Out of curiosity I installed ksysguard but can't see per-process network usage.

**Comment by thibaultmol** [comment]
? This is network activity though, right?
[image: image.png]
(make sure you're in the 'process table' tab)
